ipatests/test_integration/test_trust.py

```diff
@@ -992,3 +1052,242 @@ def test_server_option_with_unreachable_ad(self):
+        assert re.search(gid_reg, result.stdout_text)
+        assert re.search(grp_reg, result.stdout_text)
+
+    def test_idoverride_with_auto_private_group_true(self):
+        """
+        override ad trusted user in default trust view
+        and set auto-private-groups=true and ensure that
+        overridden values takes effect
+        """
+        domainusers = 'domain users'
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+        self.master.run_command(
+            [
+                "ipa",
+                "idoverrideuser-mod",
+                "Default Trust View",
+                "--uid=88888888",
+                "--gid=88888888",
+                nonposixuser,
+            ]
+        )
+        tasks.clear_sssd_cache(self.master)
+        self.mod_idrange_auto_private_group(self.existing_rangename, "true")
+        result = self.master.run_command(['id', nonposixuser])
+        uid_reg = (
+            r"^uid=(88888888)\({0}\)".format
+            (nonposixuser)
+        )
+        gid_reg = r"gid=(88888888)"
+        grp_reg = (
+            r"groups=(88888888),(\d+)\({0}@{1}\)".format
+            (domainusers, self.ad_domain)
+        )
+        assert re.search(uid_reg, result.stdout_text)
+        assert re.search(gid_reg, result.stdout_text)
+        assert re.search(grp_reg, result.stdout_text)
+
+    def test_idoverride_with_auto_private_group_hybrid(self):
+        """
+        override ad trusted user in default trust view
+        and set auto-private-groups=hybrid and ensure that
+        overridden values takes effect.
+        """
+        domainusers = 'domain users'
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+        self.master.run_command(
+            [
+                "ipa",
+                "idoverrideuser-mod",
+                "Default Trust View",
+                "--uid=78888888",
+                "--gid=78888888",
+                nonposixuser
+            ]
+        )
+        tasks.clear_sssd_cache(self.master)
+        self.mod_idrange_auto_private_group(self.existing_rangename, "hybrid")
+        result = self.master.run_command(['id', nonposixuser])
+        uid_reg = (
+            r"^uid=(78888888)\({0}\)".format
+            (nonposixuser)
+        )
+        gid_reg = r"gid=(78888888)"
+        grp_reg = (
+            r"groups=(78888888),(\d+)\({0}@{1}\)".format
+            (domainusers, self.ad_domain)
+        )
+        assert re.search(uid_reg, result.stdout_text)
+        assert re.search(gid_reg, result.stdout_text)
+        assert re.search(grp_reg, result.stdout_text)
+
+        # Remove trust
+        self.remove_trust(self.ad)
+
+
+class TestPosixAutoPrivateGroup(BaseTestTrust):
+
+    """
+    Tests for auto-private-groups option with non posix AD trust
+    """
+    topology = 'line'
+    num_ad_domains = 1
+
+    @classmethod
+    def install(cls, mh):
+        if not cls.master.transport.file_exists('/usr/bin/rpcclient'):
```

Comment:
Why is the install method overriden instead of re-using the install method from BaseTestTrust? IMO this method definition can be removed.